tests/model_registry/conftest.py

```diff
@@ -143,34 +144,130 @@ def model_registry_db_deployment(
         yield mr_db_deployment
 
 
+def create_model_registry_instance(
+    admin_client: DynamicClient,
+    namespace: str,
+    name: str,
+    labels: dict[str, str],
+    grpc: Dict[str, Any],
+    rest: Dict[str, Any],
+    istio: Dict[str, Any] | None = None,
+    oauth_proxy: Dict[str, Any] | None = None,
+    mysql: Dict[str, Any] | None = None,
+    postgres: Dict[str, Any] | None = None,
+    downgrade_db_schema_version: int | None = None,
+    enable_database_upgrade: bool | None = None,
+    wait_for_resource: bool = True,
+) -> ModelRegistry:
+    """
+    Factory function to create a ModelRegistry instance with the given parameters.
+
+    Args:
+        admin_client: The Kubernetes client to use
+        namespace: The namespace to create the model registry in
+        name: The name of the model registry instance
+        labels: Labels for the model registry
+        grpc: gRPC configuration
+        rest: REST configuration
+        istio: Optional Istio configuration
+        oauth_proxy: Optional OAuth proxy configuration
+        mysql: Optional MySQL configuration
+        postgres: Optional PostgreSQL configuration
+        downgrade_db_schema_version: Optional database downgrade schema version
+        enable_database_upgrade: Optional flag to enable database migration
+        wait_for_resource: Whether to wait for the resource to be ready
+
+    Returns:
+        A configured ModelRegistry instance
+    """
+    return ModelRegistry(
+        name=name,
+        namespace=namespace,
+        label=labels,
+        grpc=grpc,
+        rest=rest,
+        istio=istio,
+        oauth_proxy=oauth_proxy,
+        mysql=mysql,
+        postgres=postgres,
+        downgrade_db_schema_version=downgrade_db_schema_version,
+        enable_database_upgrade=enable_database_upgrade,
+        wait_for_resource=wait_for_resource,
+    )
+
+
 @pytest.fixture(scope="class")
 def model_registry_instance(
     admin_client: DynamicClient,
     model_registry_namespace: str,
     model_registry_db_deployment: Deployment,
     model_registry_db_secret: Secret,
-    model_registry_db_service: Service,
 ) -> Generator[ModelRegistry, Any, Any]:
-    with ModelRegistry(
-        name=MR_INSTANCE_NAME,
+    """Creates a model registry instance with service mesh configuration."""
+    labels = {
+        Annotations.KubernetesIo.NAME: MR_INSTANCE_NAME,
+        Annotations.KubernetesIo.INSTANCE: MR_INSTANCE_NAME,
+        Annotations.KubernetesIo.PART_OF: MR_OPERATOR_NAME,
+        Annotations.KubernetesIo.CREATED_BY: MR_OPERATOR_NAME,
+    }
+
```

Comment:
Can this be a separate fixture? We don't expect it to change. So it can be session/package scoped.

Reply:
I think rather than a fixture this would make more sense as a constant